### PR TITLE
Cheatspawn menu overhaul

### PR DESCRIPTION
**Pull request by The-Balthazar**
Porting across my debug spawn menu changes from here https://github.com/The-Balthazar/BrewLAN/commit/23479f430ee98cd27795c3f091a93dd24a7a221d

### CODE REVIEW

**Comment by Garanas** [comment]
@The-Balthazar there's still a few issues with this, do you have time to look into it?

**Comment by The-Balthazar** [comment]
What are the issues?

**Review by Garanas — APPROVED** [review]

File: lua/SimCallbacks.lua [hunk 1]
```diff
@@ -420,6 +420,8 @@ local function SetWorldCameraToUnitIconAngle(location, zoom)
 end
 
 Callbacks.ClearSpawnedMeshes = function()
+    if not CheatsEnabled() then return end
```
Comment (The-Balthazar) [inline lua/SimCallbacks.lua:423]: Unless there's a way to spawn mesh entities that get included in SpawnedMeshes without cheating already enabled, this just stops it looping over an empty table.

Comment (Garanas) [reply lua/SimCallbacks.lua:423]: I'm aware - but it is an interaction from a cheat menu. Least we can do is let people know

**Pull request merged**